webapp-django/crashstats/crashstats/static/crashstats/js/socorro/oauth2.js

```diff
@@ -110,8 +122,16 @@ var OAuth2 = (function() {
                         $.post(url, data)
                         .done(function(response) {
                             // It worked!
-                            // TODO: https://bugzilla.mozilla.org/show_bug.cgi?id=1283296
-                            document.location.reload();
+                            var next = getParameterByName('next');
+                            if (next) {
+                                // A specific URL exits.
+                                // This is most likely the case when you tried
+                                // to access a privileged URL whilst being
+                                // anonymous and being redirected.
+                                document.location.href = next;
```

Comment:
What did you have in mind? Check that it starts with a `/`? Or check that it doesn't contain any query string characters like `?` or `&`?
If I can trick your browser to visit `https://crash-stats.m.c/login/?next=/admin/some/dangerouspage/that/does/writes` then your browser would go to that after you sign in. But it's only doing a GET to get there. 

We could also whitelist a bunch of paths that are OK. Like `['/admin/', '/api/tokens/', '/profile/', '/symbols/']`.